MantidQt/API/src/QwtWorkspaceSpectrumData.cpp

```diff
@@ -75,26 +69,17 @@ Return the y value of data point i
 @return y Y value of data point i
 */
 double QwtWorkspaceSpectrumData::getY(size_t i) const {
-  double tmp = i < m_Y.size() ? m_Y[i] : m_Y.back();
-  if (m_isDistribution) {
-    tmp /= fabs(m_X[i + 1] - m_X[i]);
-  }
-  return tmp;
+  return m_isDistribution ? m_hist.frequencies()[i] : m_hist.counts()[i];
```

Comment:
This looks wrong. The `frequencies()` and `counts()` methods deal with conversion between distribution and count data depending on an *internal* state of the histogram. However, the `m_isDistribution` flag is not necessarily reflecting the internal state, since it depends on what the user wants to plot.